easy_my_coop/models/subscription_request.py

```diff
@@ -91,6 +91,17 @@ def _send_confirmation_mail(self):
     #  beware if usage in investor_wallet_platform
     @api.model
     def create(self, vals):
+        if vals.get("is_company"):
+            if not vals.get("company_name"):
+                raise ValidationError(_("Company name must be defined for a company"))
+        else:
+            if not vals.get("firstname") or not vals.get("lastname"):
```

Comment:
I find this more pythonic but it's a matter of taste

```py
if "firstname" not in vals or "lastname" not in vals:
```

Reply:
they could be in vals but as falsy values, which should also fail.

Reply:
good point 👌